lib/molinillo/resolution.rb

```diff
@@ -230,8 +239,13 @@ def state_index_for_unwind
         index = -1
         maximal_index = states.size - 2
         conflicts.each do |dependency_name, conflict|
-          next unless activated.vertex_named(dependency_name)
-          conflict.requirements.values.flatten(1).uniq.each do |r|
+          requirements_to_attempt_to_relax = [
+            conflict.requirement,
+            initial_requirement_for_name(dependency_name),
+            *binding_requirements_for_conflict(conflict),
+          ].compact.uniq
+
+          requirements_to_attempt_to_relax.each do |r|
```

Comment:
is this actually a performance improvement versus considering all of the conflict requirements? I'm wondering whether this might just end up doing more work than iterating through all of the requirements (unless this is actually pruning the search space, in which case that's 👍 but probably needs a comment)